generated type not fit `exactOptionalPropertyTypes: true`

### Description

```
Error: src/client/core/params.ts(46,29): error TS2379: Argument of type '{ in: "body" | "headers" | "path" | "query"; map: string | undefined; }' is not assignable to parameter of type '{ in: Slot; map?: string; }' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
```

related code:

```ts

const buildKeyMap = (fields: FieldsConfig, map?: KeyMap): KeyMap => {
  if (!map) {
    map = new Map();
  }

  for (const config of fields) {
    if ("in" in config) {
      if (config.key) {
        map.set(config.key, {
          in: config.in,
          map: config.map,
        });
      }
    } else if (config.args) {
      buildKeyMap(config.args, map);
    }
  }

  return map;
};

```

### Reproducible example or configuration

https://stackblitz.com/edit/hey-api-client-fetch-example

### OpenAPI specification (optional)

_No response_

### System information (optional)

_No response_

--- a/packages/custom-client/src/core/params.ts
+++ b/packages/custom-client/src/core/params.ts
@@ -31,7 +31,7 @@ type KeyMap = Map<
   string,
   {
     in: Slot;
-    map?: string;
+    map?: string | undefined;
   }
 >;
 

